plugins/org.eclipse.n4js.ts/src/org/eclipse/n4js/ts/findReferences/ConcreteSyntaxAwareReferenceFinder.java

```diff
@@ -105,4 +110,99 @@ protected EObject toValidInstanceOrNull(Resource resource, Predicate<URI> target
 		}
 		return result;
 	}
+
+	/**
+	 * This method overrides Xtext's standard behavior because we need special handling for composed members.
+	 */
+	@Override
+	protected void findLocalReferencesFromElement(
+			Predicate<URI> targetURIs,
+			EObject sourceCandidate,
+			Resource localResource,
+			Acceptor acceptor) {
```

Comment:
at this point: make sure we do not traverse the TModule if in a N4JSResource
e.g.
```
		if (sourceCandidate instanceof TModule) {
			return;
		}
```
(but check for better location)